varcode/effect_collection.py

```diff
@@ -143,8 +143,7 @@ def drop_silent_and_noncoding(self):
         """
         Create a new EffectCollection containing only non-silent coding effects
         """
-        return self.filter(
-            lambda effect: isinstance(effect, NonsilentCodingMutation))
+        return self.filter(lambda effect: effect.modifies_protein_sequence)
```

Comment:
That property defaults to False on the base class `MutationEffect` and gets overridden as True on `NonsilentCodingMutation`. The main change in this PR is adding a dynamic property to `ExonicSpliceSite` which checks its alternate effect  -- before that change `effect.modifies_protein_sequence` and `isinstance(effect, NonsilentCodingMutation)` meant the same thing.